### PR TITLE
Improve Spark Tests Framework

### PR DESCRIPTION
**Pull request by moderakh**
This PR does not make any changes to the spark source code logic. All changes are scoped to the tests of spark connector.

1. Improves tests framework. 
-- Provides common test utils for cleanup resources (Cosmos DB database, spark session, etc) after tests
-- provides framework for sharing CosmosClient, spark session, container in the scope of one test class
-- improves test framework to rely on scalatest assertions instead of assertj as a pattern
 
2. adds a Dev-Readme.md file documenting the test execution instructions.

### CODE REVIEW

File: sdk/cosmos/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+*.log
+.gitignore
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/.gitignore:3]: Why? Please remove

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/.gitignore [hunk 2]
```diff
@@ -0,0 +1,5 @@
+*.log
+.gitignore
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-0_2-12/.gitignore:2]: Why? Please remove

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/Dev-README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+```
+mvn -e -Dgpg.skip -Dmaven.javadoc.skip=true -Dspotbugs.skip=true -Dcheckstyle.skip=true -Drevapi.skip=true -pl ,azure-cosmos-spark_3-0_2-12 test package -Punit
+```
+
+To run integration tests (requires Cosmos DB endpoint)
+
+Create the file ~/cosmos-v4.properties with the following content (modify to match your cosmos endpoint):
+
+```
+ACCOUNT_HOST=https://192.168.1.51:8081
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-0_2-12/Dev-README.md:20]: IMO we should start with the deafult localhost here - https://localhost:8081

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/test/scala/com/azure/cosmos/spark/CosmosCatalogSpec.scala [hunk 4]
```diff
@@ -3,160 +3,94 @@
...
-
-    val databaseName = RandomStringUtils.randomAlphabetic(5).toLowerCase
+  it can "create a table with customized properties" taggedAs (RequiresCosmosEndpoint) in {
+    val databaseName = getAutoCleanableDatabaseName()
     val containerName = RandomStringUtils.randomAlphabetic(6).toLowerCase + System.currentTimeMillis()
 
-    spark.sql(s"CREATE DATABASE cosmoscatalog.${databaseName};")
-    spark.sql(s"CREATE TABLE cosmoscatalog.${databaseName}.${containerName} (word STRING, number INT) using cosmos.items " +
+    spark.sql(s"CREATE DATABASE testCatalog.${databaseName};")
+    spark.sql(s"CREATE TABLE testCatalog.${databaseName}.${containerName} (word STRING, number INT) using cosmos.items " +
```
Comment (FabianMeiswinkel) [inline sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/test/scala/com/azure/cosmos/spark/CosmosCatalogSpec.scala:84]: Unrelated - but I really love that you have wired up the metadata changes already - I am sure Spark developers will love this feature - don't know how many customers I have seen that had taken a dependency to some crappy very old Python SDK just to be able to create/modify containers etc.

**Review by FabianMeiswinkel — APPROVED** [review]
Thanks! LGTM - except for minor comments. On .gitignore this is mostly a question - if there is a reason why you added .gitignore to the .gitgnore file I missed feel free to ignore

File: sdk/cosmos/.gitignore (hunk 1 above)
Comment (moderakh) [reply sdk/cosmos/.gitignore:3]: good catch. will remove.

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/.gitignore (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-0_2-12/.gitignore]: good catch. removed in the next commit.

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/Dev-README.md (hunk 3 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-0_2-12/Dev-README.md]: good catch. changed to localhost. thanks.

**Comment by moderakh** [comment]
> Thanks! LGTM - except for minor comments. On .gitignore this is mostly a question - if there is a reason why you added .gitignore to the .gitgnore file I missed feel free to ignore

Thanks for catching the .gitignore files. I just removed the .gitignore entry in the .gitignore, that shouldn't be needed.

**Pull request merged**